std.go

```
@@ -26,30 +49,58 @@ func (a *StdManagerAdapter[T]) On(name string, listener Listener[T], priority ..
 		if !ok {
 			var zeroT T
-			return fmt.Errorf("event: data type mismatch for event '%s'. Listener expected type %T, but event data is type %T", e.Name(), zeroT, e.Data())
+			return fmt.Errorf("event: data type mismatch for event '%s'. Listener expected type %T, but event data is type %T",
+				e.Name(), zeroT, e.Data())
 		}
+
+		if once {
+			switch v := mgr.(type) {
+			case EventManager[T]:
+				v.RemoveListener(e.Name(), listener)
+			case *StdManagerAdapter[T]:
```

golangci-lint (1.64.8)

SA4020: unreachable case clause: go.lumeweb.com/event/v2.EventManager[T] will always match before *go.lumeweb.com/event/v2.StdManagerAdapter[T]

(staticcheck)